plugins/auth-backend-module-microsoft-provider/src/authenticator.ts

```diff
@@ -21,21 +21,30 @@ import {
   PassportProfile,
 } from '@backstage/plugin-auth-node';
 import { ExtendedMicrosoftStrategy } from './strategy';
-import { union } from 'lodash';
 
 /** @public */
 export const microsoftAuthenticator = createOAuthAuthenticator({
   defaultProfileTransform:
     PassportOAuthAuthenticatorHelper.defaultProfileTransform,
+  scopes: {
+    required: ['email', 'openid', 'offline_access', 'user.read'],
+    transform({ requested, granted, required, additional }) {
+      // Resources scopes are of the form `<resource>/<scope>`, and are handled
+      // separately from the normal scopes in the client. When request a
+      // resource scope we should only include forward the request scope along
+      // with offline_access.
+      const hasResourceScope = Array.from(requested).some(s => s.includes('/'));
```

Comment:
Did some testing locally.
The additionalScopes config value gets passed into 'additional' not 'requested'

This is tested as working when directly modifying the node_module locally, small change:
```
const hasResourceScope = Array.from(additional).some((s) => s.includes("/"));
if (hasResourceScope) {
      return [...additional, "offline_access"];
}
```

Reply:
@Rugvip 
Should I make an issue about this?